schematypes.go

```diff
@@ -1488,102 +1509,296 @@ type DisassembledInstruction struct {
-func (*ContinueRequest) isMessage()                 {}
-func (*ContinueResponse) isMessage()                {}
-func (*NextRequest) isMessage()                     {}
-func (*NextResponse) isMessage()                    {}
-func (*StepInRequest) isMessage()                   {}
-func (*StepInResponse) isMessage()                  {}
-func (*StepOutRequest) isMessage()                  {}
-func (*StepOutResponse) isMessage()                 {}
-func (*StepBackRequest) isMessage()                 {}
-func (*StepBackResponse) isMessage()                {}
-func (*ReverseContinueRequest) isMessage()          {}
-func (*ReverseContinueResponse) isMessage()         {}
-func (*RestartFrameRequest) isMessage()             {}
-func (*RestartFrameResponse) isMessage()            {}
-func (*GotoRequest) isMessage()                     {}
-func (*GotoResponse) isMessage()                    {}
-func (*PauseRequest) isMessage()                    {}
-func (*PauseResponse) isMessage()                   {}
-func (*StackTraceRequest) isMessage()               {}
-func (*StackTraceResponse) isMessage()              {}
-func (*ScopesRequest) isMessage()                   {}
-func (*ScopesResponse) isMessage()                  {}
-func (*VariablesRequest) isMessage()                {}
-func (*VariablesResponse) isMessage()               {}
-func (*SetVariableRequest) isMessage()              {}
-func (*SetVariableResponse) isMessage()             {}
-func (*SourceRequest) isMessage()                   {}
-func (*SourceResponse) isMessage()                  {}
-func (*ThreadsRequest) isMessage()                  {}
-func (*ThreadsResponse) isMessage()                 {}
-func (*TerminateThreadsRequest) isMessage()         {}
-func (*TerminateThreadsResponse) isMessage()        {}
-func (*ModulesRequest) isMessage()                  {}
-func (*ModulesResponse) isMessage()                 {}
-func (*LoadedSourcesRequest) isMessage()            {}
-func (*LoadedSourcesResponse) isMessage()           {}
-func (*EvaluateRequest) isMessage()                 {}
-func (*EvaluateResponse) isMessage()                {}
-func (*SetExpressionRequest) isMessage()            {}
-func (*SetExpressionResponse) isMessage()           {}
-func (*StepInTargetsRequest) isMessage()            {}
-func (*StepInTargetsResponse) isMessage()           {}
-func (*GotoTargetsRequest) isMessage()              {}
-func (*GotoTargetsResponse) isMessage()             {}
-func (*CompletionsRequest) isMessage()              {}
-func (*CompletionsResponse) isMessage()             {}
-func (*ExceptionInfoRequest) isMessage()            {}
-func (*ExceptionInfoResponse) isMessage()           {}
-func (*ReadMemoryRequest) isMessage()               {}
-func (*ReadMemoryResponse) isMessage()              {}
-func (*DisassembleRequest) isMessage()              {}
-func (*DisassembleResponse) isMessage()             {}
+func (m *Request) GetSeq() int          { return m.Seq }
```

Comment:
You seem to have lost ProtocolMessage here. It should be part of the interface hierarchy, no? Aren't there some tests that rely on using ProtocolMessage to represent unknown type bad messages and check how those would be handled?

Reply:
Logically, I don't think `ProtocolMessage` should satisfy the interface, because a `ProtocolMessage` on its own is not a valid DAP message - it's just a bunch of common fields they all have.

We had one place in Delve that used it as a message, but it was just an un-referenced test helper - I removed it in today's update of https://github.com/eliben/delve/pull/3